src/test/base32_tests.cpp

```diff
@@ -23,14 +23,14 @@ BOOST_AUTO_TEST_CASE(base32_testvectors)
 
     // Decoding strings with embedded NUL characters should fail
     bool failure;
-    (void)DecodeBase32(std::string("invalid", 7), &failure);
-    BOOST_CHECK_EQUAL(failure, true);
-    (void)DecodeBase32(std::string("AWSX3VPP", 8), &failure);
-    BOOST_CHECK_EQUAL(failure, false);
+    (void)DecodeBase32(std::string("invalid", 8), &failure);
+    BOOST_CHECK(failure);
+    (void)DecodeBase32(std::string("AWSX3VPP"), &failure);
+    BOOST_CHECK(!failure);
     (void)DecodeBase32(std::string("AWSX3VPP\0invalid", 16), &failure);
-    BOOST_CHECK_EQUAL(failure, true);
-    (void)DecodeBase32(std::string("AWSX3VPPinvalid", 15), &failure);
-    BOOST_CHECK_EQUAL(failure, true);
+    BOOST_CHECK(failure);
+    (void)DecodeBase32(std::string("AWSX3VPPinvalid", 16), &failure);
```

Comment:
Maybe we can wait with making this change for the C++17 switch? We'll have the s-suffix strings, so don't need to worry about specifying the length explicitly at all anymore even when there are trailing '\0' characters.
```c++
#include <iostream>
#include <string>

int main()
{
    using namespace std::string_literals;

    std::string x = "test\x00\x00\x00\x00"s;
    std::cout << x.size() << std::endl; // prints 8
}
```

Reply:
I think this PR is good as is, but if it is still unmerged when we start allowing C++17, then I will update it to use `"..."s`;

Reply:
> then I will update it to use "..."s;

Even though this has a few ACKs, I think you can go ahead and update it to use the s-suffix strings. We'll merge this post branch off when requiring C++17.

Reply:
Done